src/main/java/org/alliancegenome/curation_api/dao/AGMDiseaseAnnotationDAO.java

```diff
@@ -15,9 +15,9 @@ protected AGMDiseaseAnnotationDAO() {
 		super(AGMDiseaseAnnotation.class);
 	}
 
-	public List<Long> findAllAnnotationIds(String taxonID) {
-		Query jpqlQuery = entityManager.createQuery("SELECT annotation.id FROM AGMDiseaseAnnotation annotation WHERE annotation.subject.taxon.curie=:taxonId");
-		jpqlQuery.setParameter("taxonId", taxonID);
+	public List<Long> findAllAnnotationIds(String speciesName) {
+		Query jpqlQuery = entityManager.createQuery("SELECT annotation.id FROM AGMDiseaseAnnotation annotation WHERE annotation.subject.taxon.name LIKE CONCAT(:speciesName, '%')");
+		jpqlQuery.setParameter("speciesName", speciesName);
```

Comment:
Doing a like query is going to be slow vs using a id field.

Reply:
Sure, but we don’t know all the IDs (and it may not be just 1) unless we do another query and this query is only run once per load